### PR TITLE
Guild profs

### PR DESCRIPTION
**Pull request by AlanMWalker**

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -37,51 +51,270 @@ let commandDescriptionArray = {
...
 function runTest(msg) {
-    msg.channel.send({
-        embed: {
-            color: MSGCOLOUR,
-            title: 'Test title',
-            description: 'description'
+
+}
+
+// #region GuildProfs functions
```
Comment (AlanMWalker) [inline main.js]: Is this meant to be left in?

File: main.js [hunk 2]
```diff
@@ -37,51 +51,270 @@ let commandDescriptionArray = {
...
+
+// Used to list professions by player or prof
+// !gp <player|prof> <name>
+function runListGuildProfs(msg, params) {
```
Comment (AlanMWalker) [inline main.js]: Function length seems long, can it be split any further?

File: utilitys.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+exports.identify = function (msg) {
+    msg.channel.send(os.userInfo().username);
+}
+
+exports.loadFile = function (file, dataType) {
+    let out = fs.readFileSync(file);
+
+    if (dataType == 'json') {
+        return JSON.parse(out);
+    } else {
```
Comment (AlanMWalker) [inline utilitys.js]: The else is redundant, can just be an if for the json, and return out by default. Also lacks any checking for undefined return type from readFileSync.

File: main.js (hunk 1 above)
Comment (ZackCrichton) [reply main.js]: Yes, is part of runTest command. should be left in.

File: main.js (hunk 2 above)
Comment (ZackCrichton) [reply main.js]: Could split list player and list prof into their own functions. Discussion of further optimisation would be good.

File: utilitys.js (hunk 3 above)
Comment (ZackCrichton) [reply utilitys.js]: Understood, changes will be made.

File: main.js (hunk 2 above)
Comment (ZackCrichton) [reply main.js]: Could also pull all guild prof functions into their own file

**Pull request merged**